stdlib/marshal.mli

```diff
@@ -109,6 +109,9 @@ val to_channel : out_channel -> 'a -> extern_flags list -> unit
    when read back on a 32-bit platform.  The [Mashal.Compat_32] flag
    only matters when marshaling is performed on a 64-bit platform;
    it has no effect if marshaling is performed on a 32-bit platform.
+   
+   It raises [Failure "output_value: not a binary channel"] if channel
+   [chan] is not in binary mode.
```

Comment:
The consensus is that we should not be explicit about the string message in the specification of the function, to be allowed to change it in the future. So maybe just say that a `Failure` exception is raised?

Reply:
Agreed! I took the doc from `Marshal.from_channel` as blue print, so we might omit the message string there as well. Do you prefer `Failure` or `Failure _` as suggested by @nojb ?